server/server.go

```diff
@@ -552,9 +553,26 @@ func (s *Server) startMetricsServer() error {
 	}
 
 	if err := startMetricsEndpoint(
-		"tcp", fmt.Sprintf(":%v", s.config.MetricsPort), me,
+		"tcp", fmt.Sprintf(":%v", s.config.MetricsPort), "", "", me,
```

Comment:
should we be checking if this is 0, and not add the endpoint if so?

Reply:
Are you referring to the port? We may add this to the config validation:

- Provided cert but no key is invalid
- Provided key but no cert is invalid
- Provided cert and key but port is 0 is invalid

Maybe add a check if the file exists on disk.

Reply:
I more meant: we have two different ports configurable. if a user configures one with TLS, would they want the unsecured one? currently, we don't have a way to disable the unsecured one. is this something we want?

Reply:
Alright, removed the secure port and used the usual port for it. If cert and key are provided, then we will not serve the metrics as insecure any more.